docs/DeveloperGuide.adoc

```diff
@@ -176,94 +176,214 @@ through all the stored data and generate the statistics data whenever it is quer
 ** Cons: Requires managing the storage of the statistical data and possibly lead to lowered performance of other features such as the command which does CRUD to Employees/Events which will be slower with the need to generate the statistical data and store it in the JSON file storage.
 // end::statistics[]
 
-// tag::undoredo[]
-=== [Proposed] Undo/Redo feature
-==== Proposed Implementation
+=== Auto allocation feature
+==== Implementation
+
+The `AutoAllocateCommand` has a auto-allocation mechanism which is facilitated by methods in `Event`.
+The `AutoAllocateCommand` takes in three arguments:
 
-The undo/redo mechanism is facilitated by `VersionedAddressBook`.
-It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`.
-Additionally, it implements the following operations:
+1.  `eventIndex` - index of event in the displayed event list
+2. `ManpowerCountToAdd` - number of employees to allocate [optional]
+3. `tagList` - a set of tags to filter the employees [optional]
 
-* `VersionedAddressBook#commit()` -- Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` -- Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` -- Restores a previously undone address book state from its history.
+Additionally, the `AutoAllocateCommand` uses the following operations:
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+* `Event#isAvailableForEvent()` -- Checks if an employee is available for the event.
+* `Event#allocateEmployee()` -- Checks if an employee is available for the event.
+* `AutoAllocateCommand#createAvailableEmployeeListForEvent()` -- Creates a list of employees available for the event,
+filtered by the tags specified by user.
+* `AutoAllocateCommand#getManpowerNeededByEvent()` -- Calculates the number of employees currently required
+by the event.
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+Given below is an example usage scenario and how the auto allocation mechanism behaves at each step.
```

Comment:
You won't get any credit for this because it's not a UML diagram. Since this is a flowchart-style diagram, maybe you can consider making it an activity diagram? I think you could easily make it into one by adding start and end points.

Reply:
Thanks for the suggestion!